src/DynamoCoreWpf/ViewModels/Core/WorkspaceViewModel.cs

```diff
@@ -1560,19 +1578,13 @@ private void RefreshViewOnSelectionChange(object sender, NotifyCollectionChanged
             RaisePropertyChanged("SelectionArgumentLacing");            
         }
 
-        /// <summary>
-        /// Returns ViewModelBase by GUID
-        /// </summary>
-        /// <param name="modelGuid">Identifier of the requested model.</param>
-        /// <returns>Found <see cref="ViewModelBase"/> object.</returns>
-        internal ViewModelBase GetViewModelInternal(Guid modelGuid)
+        internal ViewModelBase GetViewModel(Guid modelGuid)
         {
-            ViewModelBase foundModel = (Connectors.FirstOrDefault(c => c.ConnectorModel.GUID == modelGuid)
-                ?? Nodes.FirstOrDefault(node => node.NodeModel.GUID == modelGuid) as ViewModelBase)
-                ?? (Notes.FirstOrDefault(note => note.Model.GUID == modelGuid)
-                ?? Annotations.FirstOrDefault(annotation => annotation.AnnotationModel.GUID == modelGuid) as ViewModelBase);
-
-            return foundModel;
+            return ViewModelMap.TryGetValue(modelGuid, out ViewModelBase viewModel) ? viewModel : null;
+        }
+        internal VmType GetViewModel<VmType>(Guid modelGuid) where VmType : ViewModelBase
```

Comment:
How about renaming this to `GetViewModelAs`?

Reply:
sure